### PR TITLE
Movement refinements

### PR DESCRIPTION
**Pull request by cadenceblorbo**
-slowed down character overall
-changed glide so that it automatically occurs if jump is held while falling, and limit glide turning speed
-allowed high jump to be triggered by buffered jumps
-change how horizontal glide speed is calculated to better reward high fall speed (dive->glide, advanced tech)
-some refactoring

### CODE REVIEW

**Review by jidaprano — COMMENTED** [review]
- Code quality looks good to me.
- During testing I did run into the weirdness that occurs when you collide with the very corner where the top of a platform meets the side; this was especially evident if I was in a glide at the time. I just got jittered around a bit then either got to the top or fell down somewhat unpredictably. However, after accidentally running into this once or twice I did have to try pretty hard to reproduce it again so it's probably not a huge deal unless you want to implement a mantle kind of thing or something.
- When doing walk->jump off edge->fall the animations work but when doing walk->jump off edge->glide->fall the animation gets stuck in glide and doesn't switch back to fall if shift is released during the glide.
- Ran into a bit of getting stuck on corners of objects when going up a sloped surface. 
- Other than those things, it feels exquisite; the roll->speed up process is a lot more incentivized with the movement being a bit slower and the falling/gliding/plunging feels very good too.

File: Assets/Scripts/CharacterMechanics/AdvancedMovementStates/Deceleration.cs [hunk 1]
```diff
@@ -22,7 +22,13 @@ void Start()
             { AdvancedMovementState.MoveStopping, controlLifted },
             { AdvancedMovementState.Rolling, pushedActionButton && Movement.IsOnGround() },
             { AdvancedMovementState.Plunging, pushedActionButton && !Movement.IsOnGround() },
-            { AdvancedMovementState.Gliding, jumpedOffGround },
+            {
+                AdvancedMovementState.Gliding,
+                Movement.Jump.IsPressed()
+                    && !Movement.IsOnStableGround()
+                    && Movement.ExtraJumpsRemaining == 0
```
[the remark this answers is not in the document]
Comment (cadenceblorbo) [reply Assets/Scripts/CharacterMechanics/AdvancedMovementStates/Deceleration.cs:29]: We are not planning on double jump, but keeping this in doesn't break stuff and _should_ allow us to add one later.

**Comment by cadenceblorbo** [comment]
> * Code quality looks good to me.
> 
>     * During testing I did run into the weirdness that occurs when you collide with the very corner where the top of a platform meets the side; this was especially evident if I was in a glide at the time. I just got jittered around a bit then either got to the top or fell down somewhat unpredictably. However, after accidentally running into this once or twice I did have to try pretty hard to reproduce it again so it's probably not a huge deal unless you want to implement a mantle kind of thing or something.
> 
>     * When doing walk->jump off edge->fall the animations work but when doing walk->jump off edge->glide->fall the animation gets stuck in glide and doesn't switch back to fall if shift is released during the glide.
> 
>     * Ran into a bit of getting stuck on corners of objects when going up a sloped surface.
> 
>     * Other than those things, it feels exquisite; the roll->speed up process is a lot more incentivized with the movement being a bit slower and the falling/gliding/plunging feels very good too.

All the mentioned bugs will be addressed when Paige does her movement changes I think, this PR is just my stuff which dealt with polishing overall feel.

**Comment by jidaprano** [comment]
Sounds good to me; should be good to merge then.

**Pull request merged**